processing/src/main/java/io/druid/query/filter/InDimFilter.java

```diff
@@ -117,6 +121,46 @@ public ExtractionFn getExtractionFn()
   @Override
   public DimFilter optimize()
   {
+    InDimFilter inFilter = optimizeLookup();
+    if (inFilter.values.size() == 1) {
+      return new SelectorDimFilter(inFilter.dimension, inFilter.values.first(), inFilter.getExtractionFn());
+    }
+    return inFilter;
+  }
+
+  public InDimFilter optimizeLookup() {
```

Comment:
minor nit : should this be private so that users of this class always use optmize instead of calling optimizeLookup() ?